### PR TITLE
feat(remix-server-runtime): pass `AppLoadContext` to `handleRequest`

### PR DESCRIPTION
**Pull request by fernandojbf**
This is the same feature change request done [here](https://github.com/remix-run/remix/pull/2045). A new [discussion](https://github.com/remix-run/remix/discussions/5314) was created to this.

My action was mainly check the changes by introduced @axel-habermaier's PR and apply them again. 

This is my first contribution, if i made something wrong please let me know.

Closes: #

- [x] Docs
- [x] Tests

Testing Strategy:

I've added unit tests on server-runtime passing loadContext to the handler.

### CODE REVIEW

**Comment by fernandojbf** [comment]
Bahh, I've made a rebase from main :( i will fix that in a moment

**Review by jacob-ebey — APPROVED** [review]

File: .changeset/twelve-cheetahs-drive.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"remix": patch
+"@remix-run/serve": patch
+"@remix-run/server-runtime": patch
+---
+
+feat(remix-server-runtime): pass `AppLoadContext` to `handleRequest`
```
Comment (MichaelDeBoey) [inline .changeset/twelve-cheetahs-drive.md:7]:
```suggestion
pass `AppLoadContext` to `handleRequest`
```

File: .changeset/twelve-cheetahs-drive.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+"remix": patch
+"@remix-run/serve": patch
```
Comment (MichaelDeBoey) [inline .changeset/twelve-cheetahs-drive.md:3]:
```suggestion
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

**Review by MichaelDeBoey — APPROVED** [review]

**Comment by kiliman** [comment]
I, too, would like to see the PR merged. I think it was an oversight in the initial implementation. The other other `handle*Request` functions contain the context (as they came later).

**Comment by fernandojbf** [comment]
Thank you 🎉

**Comment by jacob-ebey** [comment]
> Thank you 🎉

Thank YOU! :)

**Pull request merged**